app/views/contexts/section_summary_context.py

```diff
@@ -65,15 +65,17 @@ def _build_summary(self, location):
 
     def _title_for_location(self, location):
 
-        title = self._schema.get_section(location.section_id).get("title")
-        if location.list_item_id:
-            repeating_title = self._schema.get_repeating_title_for_section(
-                location.section_id
+        section_id = location.section_id
+        title = (
+            self._schema.get_repeating_title_for_section(section_id)
+            or self._schema.get_summary_title_for_section(section_id)
+            or self._schema.get_title_for_section(section_id)
+        )
```

Comment:
Could be wrapped in a `get_section_summary_title_for_section` method and in future cached.